examples/monad.kl

```diff
@@ -0,0 +1,231 @@
+#lang "prelude.kl"
+
+(import "defun.kl")
+(import "lambda-case.kl")
+(import (shift "prelude.kl" 1))
+(import (shift "let.kl" 1))
+(import (shift "quasiquote.kl" 1))
+
+
+(example (lambda (x) x))
+
+(datatype (Functor F A B)
+  (functor (-> (-> A B) (-> (F A) (F B)))))
```

Comment:
Hmm, that's not quite the definition I expected! I am guessing that we cannot yet have a field of type `forall a b. something`?